model/infinispan/src/main/java/org/keycloak/crl/infinispan/InfinispanCrlStorageProviderFactory.java

```diff
@@ -0,0 +1,115 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.keycloak.crl.infinispan;
+
+import java.security.cert.X509CRL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.FutureTask;
+import org.infinispan.Cache;
+import org.keycloak.Config;
+import org.keycloak.connections.infinispan.InfinispanConnectionProvider;
+import org.keycloak.crl.CrlStorageProvider;
+import org.keycloak.crl.CrlStorageProviderFactory;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.provider.ProviderConfigProperty;
+import org.keycloak.provider.ProviderConfigurationBuilder;
+
+/**
+ *
+ * @author rmartinc
+ */
+public class InfinispanCrlStorageProviderFactory implements CrlStorageProviderFactory {
+
+    public static final String PROVIDER_ID = "infinispan";
+
+    private volatile Cache<String, X509CRLEntry> crlCache;
+    private final Map<String, FutureTask<X509CRL>> tasksInProgress = new ConcurrentHashMap<>();
+    private int cacheTime;
+    private int minTimeBetweenRequests;
+
+    @Override
+    public CrlStorageProvider create(KeycloakSession session) {
+        lazyInit(session);
+        return new InfinispanCrlStorageProvider(crlCache, tasksInProgress, cacheTime, minTimeBetweenRequests);
+    }
+
+    @Override
+    public List<ProviderConfigProperty> getConfigMetadata() {
+        return ProviderConfigurationBuilder.create()
+                .property()
+                    .name("cacheTime")
+                    .type("int")
+                    .helpText(
+                            """
+                            Interval in seconds that the CRL is cached. The next update time of the CRL is always a minimum if present.
+                            Zero or a negative value means CRL is cached until the next update time specified in the CRL (or inifinite if the
+                            CRL does not contain the next update).
+                            """
+                    )
+                    .defaultValue(-1)
+                    .add()
+                .property()
+                    .name("minTimeBetweenRequests")
+                    .type("int")
+                    .helpText(
+                            """
+                            Minimum interval in seconds between two requests to retrieve the CRL. The CRL is not updated
+                            from the URL again until this minimum time has passed since the previous refresh. In theory
+                            this option is never used if the CRL is refreshed correctly in the next update time.
+                            Default 10 seconds.
```

Comment:
is a zero or a negative value valid? Probably worth adding a sentence about the behavior with a zero or negative value.